### PR TITLE
Adds the ability to enable a redirect for a given storage provider

### PR DESCRIPTION
**Pull request by robbo10**
Fixes the following issue  #15 

Signed-off-by: Kieran Robinson <krobn@allstate.com>

### CODE REVIEW

File: jobs/harbor/templates/bin/pre-start.erb [hunk 1]
```diff
@@ -51,6 +51,10 @@ echo "$ip $fqdn" >> /etc/hosts
 <%-   end %>
 <%- end %>
 
+<%- if p("registry_storage_provider.redirect") %>
+sed -i -e '/storage:$/a \ \ redirect:\n \ \ \ \disable: true' /var/vcap/jobs/harbor/packages/harbor-app/common/templates/registry/config.yml
```
[the remark this answers is not in the document]
Comment (robbo10) [reply jobs/harbor/templates/bin/pre-start.erb]: Hey @jessehu - we tested various scenarios with the above code and it seemed to work fine, however we can change no issues.

**Comment by robbo10** [comment]
@jessehu - we have updated per your comments sir.

**Comment by robbo10** [comment]
@jessehu - any update?

**Comment by jessehu** [comment]
@robbo10 I'm very sorry for the late response.

**Comment by jessehu** [comment]
Hi @robbo10 did you get my review comments: 
https://github.com/vmware/harbor-boshrelease/pull/16#discussion_r201609117
https://github.com/vmware/harbor-boshrelease/pull/16#discussion_r201609767

**Comment by robbo10** [comment]
@jessehu - apologies we have been very busy, we can look into this in a few days, our original PR was fully tested though and worked fine.  In the spec file the default for `registry_storage_provider.redirect` is false.

**Comment by jessehu** [comment]
Thanks @robbo10. 
1) The line 55 "if ! grep -q 'redirect: {disable: true}' ; then" has grammar error because 'grep' command requires a file to search in.
2) https://docs.docker.com/registry/configuration/#storage contains the configuration 
```
  redirect:
    disable: false
```
So I suggest naming the property as "registry_storage_provider.redirect.disable" to align with the yaml hierachy above.

**Comment by robbo10** [comment]
Hey @jessehu we have made the required changes and have fully tested all possibilities through our CI pipeline.  All the changes are working as expected.

Thanks

**Comment by jessehu** [comment]
Thanks @robbo10 . This patch looks great! +1 approval.
Copying @stonezdj to review as well.

**Comment by clouderati** [comment]
Hey @robbo10 – thanks for the PR!

**Pull request merged**